src/main/java/com/gradle/CommonCustomUserDataGradleEnterpriseListener.java

```diff
@@ -5,14 +5,14 @@
 import com.gradle.maven.extension.api.GradleEnterpriseListener;
 import org.apache.maven.MavenExecutionException;
 import org.apache.maven.execution.MavenSession;
-import org.codehaus.plexus.component.annotations.Component;
+import org.eclipse.sisu.Description;
 
-@SuppressWarnings("unused")
-@Component(
-    role = GradleEnterpriseListener.class,
-    hint = "common-custom-user-data-gradle-enterprise-listener",
-    description = "Captures common custom user data in Maven build scans"
-)
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+@Named
```

Comment:
```suggestion
@Named("common-custom-user-data-gradle-enterprise-listener")
```
ditto

Reply:
Added in c6f18739bfdc991b74ede3409d111f0254d9ecfd